runtime/description-dom-formatter.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * @license
+ * Copyright (c) 2017 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+"use strict";
+
+import assert from '../platform/assert-web.js';
+import {DescriptionFormatter} from './description.js';
+
+export default class DescriptionDomFormatter extends DescriptionFormatter {
+  constructor(description) {
+    super(description);
+    this._nextID = 0;
+  }
+
+  _isSelectedDescription(desc) {
+    return super._isSelectedDescription(desc) || (!!desc.template && !!desc.model);
+  }
+
+  _populateParticleDescription(particle, descriptionByName) {
+    let result = super._populateParticleDescription(particle, descriptionByName);
+
+    if (descriptionByName["_template_"]) {
+      result = Object.assign(result, {
+        template: descriptionByName["_template_"],
+        model: Object.assign({}, JSON.parse(descriptionByName["_model_"]))
```

Comment:
Why `Object.assign`? My expectation is that the object returned by `JSON.parse` is already fresh and clean. :)